src/Entity/IdeasWorkshop/Vote.php

```diff
@@ -32,25 +50,34 @@ class Vote
     /**
      * @ORM\ManyToOne(targetEntity="Idea", inversedBy="votes")
      * @ORM\JoinColumn(nullable=false)
+     *
+     * @Assert\NotNull
+     *
+     * @SymfonySerializer\Groups({"idea_list_read", "vote_read", "vote_write"})
      */
     private $idea;
 
     /**
      * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
      * @ORM\JoinColumn(onDelete="CASCADE", nullable=false)
+     *
+     * @Assert\NotNull
+     *
+     * @SymfonySerializer\Groups({"idea_list_read", "vote_read", "vote_write"})
      */
     private $adherent;
 
     /**
      * @ORM\Column(length=10)
      *
      * @Assert\Choice(callback={"AppBundle\Entity\IdeasWorkshop\VoteTypeEnum", "toArray"})
+     *
+     * @SymfonySerializer\Groups({"idea_list_read", "vote_read", "vote_write"})
      */
     private $type;
 
-    public function __construct(Adherent $adherent, string $type)
+    public function __construct(string $type = null)
     {
-        $this->adherent = $adherent;
         $this->type = $type;
```

Comment:
what about checking the type value with `VoteTypeEnum::isValid($type)`?

Reply:
Now, it's checked with validation while setting a value https://github.com/EnMarche/en-marche.fr/pull/2946/files#diff-39c4e64c8fc7a47c56a13eaa32ae313aL47